src/id/shinigami/src/eu/kanade/tachiyomi/extension/id/shinigami/Shinigami.kt

```diff
@@ -83,70 +56,198 @@ class Shinigami : Madara("Shinigami", "https://shinigami09.com", "id"), Configur
-        val otherId = OTHER_ID_REGEX.findAll(script).firstOrNull { it.groupValues[1] != "post" }?.groupValues?.get(2) ?: throw Exception("Unable to get other id")
-        val key = otherId + keyMatch[1] + postId + keyMatch[2] + postId
-        val salt = chapterData.s.decodeHex()
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
 
-        val unsaltedCiphertext = Base64.decode(chapterData.ct, Base64.DEFAULT)
-        val ciphertext = salted + salt + unsaltedCiphertext
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/series/" + manga.url
+    }
 
-        val decrypted = CryptoAES.decrypt(Base64.encodeToString(ciphertext, Base64.DEFAULT), key)
-        val data = json.decodeFromString<List<String>>(decrypted)
-        return data.mapIndexed { idx, it ->
-            Page(idx, document.location(), it)
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        // Migration from old web urls to the new api based
+        if (manga.url.startsWith("/series/")) {
+            throw Exception("Migrate dari $name ke $name (ekstensi yang sama)")
         }
+
+        return GET("$apiUrl/$API_BASE_PATH/manga/detail/" + manga.url, apiHeaders)
     }
 
-    private fun randomString(length: Int): String {
-        val charPool = ('a'..'z') + ('A'..'Z')
-        return List(length) { charPool.random() }.joinToString("")
+    override fun mangaDetailsParse(response: Response): SManga {
+        val mangaDetailsResponse = response.parseAs<ShinigamiMangaDetailDto>()
+        val mangaDetails = mangaDetailsResponse.data
+
+        return SManga.create().apply {
+            author = mangaDetails.taxonomy["Author"]?.joinToString(", ") { it.name }.orEmpty()
+            artist = mangaDetails.taxonomy["Artist"]?.joinToString(", ") { it.name }.orEmpty()
+            status = mangaDetails.status.toStatus()
+            description = mangaDetails.description
+
+            val genres = mangaDetails.taxonomy["Genre"]?.joinToString(", ") { it.name }.orEmpty()
+            val type = mangaDetails.taxonomy["Format"]?.joinToString(", ") { it.name }.orEmpty()
+            genre = listOf(genres, type).filter { it.isNotBlank() }.joinToString(", ")
+        }
+    }
+
+    private fun Int.toStatus(): Int {
+        return when (this) {
+            1 -> SManga.ONGOING
+            2 -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        return GET("$apiUrl/$API_BASE_PATH/chapter/${manga.url}/list?page_size=3000", apiHeaders)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val result = response.parseAs<ShinigamiChapterListDto>()
+
+        return result.chapterList.map(::chapterFromObject)
+    }
+
+    private fun chapterFromObject(obj: ShinigamiChapterListDataDto): SChapter = SChapter.create().apply {
+        date_upload = obj.date.toDate()
```

Comment:
```suggestion
        date_upload = dateFormat.tryParse(obj.date)
```
save `SimpleDateFormat` in a class val and access it here like so, instead of creating it each time inside `toDate`